extensions/cursorrules/src/components/actions/CopyRuleAction.tsx

```diff
@@ -0,0 +1,131 @@
+import {
+  Action,
+  Clipboard,
+  Icon,
+  Keyboard,
+  LaunchType,
+  Toast,
+  confirmAlert,
+  getPreferenceValues,
+  launchCommand,
+  open,
+  showHUD,
+  showToast,
+} from "@raycast/api";
+import { crossLaunchCommand } from "raycast-cross-extension";
+import type { CursorRule } from "../../types";
+
+interface Props {
+  cursorRule: CursorRule;
+}
+
+export const CopyRuleAction = ({ cursorRule }: Props) => {
+  const preferences = getPreferenceValues<Preferences>();
+
+  async function copyAndApplyRule() {
+    try {
+      await Clipboard.copy(cursorRule.content);
+      await showSuccessToast();
+
+      if (await shouldOpenProject()) {
+        await openProject();
+      }
+    } catch (error) {
+      console.debug("Failed to copy and apply rule:", error);
+      await showErrorToast();
+    }
+  }
+
+  async function showSuccessToast() {
+    await showToast({
+      style: Toast.Style.Success,
+      title: "Rule Copied",
+      message: "Ready to paste into .cursorrules file",
+    });
+  }
+
+  async function showErrorToast() {
+    await showToast({
+      style: Toast.Style.Failure,
+      title: "Error",
+      message: "Failed to copy and apply rule",
+    });
+  }
+
+  async function shouldOpenProject(): Promise<boolean> {
+    if (preferences.skipConfirmationOnCopy) return true;
+
+    return await confirmAlert({
+      title: "Open Projects",
+      message: "Do you want to open projects to apply this rule?",
+      primaryAction: { title: "Open" },
+      dismissAction: {
+        title: "Cancel",
+        onAction: () => showHUD("Paste into .cursorrules file in your project"),
+      },
+    });
+  }
+
+  async function openProject() {
+    const context = {
+      cursorDirectory: {
+        ruleName: `${cursorRule.slug}.md`,
+        ruleContent: cursorRule.content,
+        replace: preferences.replaceOnLaunch,
+      },
+    };
+
+    if (preferences.projectsDirectory) {
+      try {
+        await launchCommand({
+          name: "projects",
+          type: LaunchType.UserInitiated,
+          context,
+        });
+      } catch (error) {
+        console.debug("Failed to launch projects command:", error);
+        await launchCursorRecentProjects(context);
+      }
+    } else {
+      await launchCursorRecentProjects(context);
+    }
+  }
+
+  return (
+    <Action
+      title="Copy and Apply Rule"
+      onAction={copyAndApplyRule}
+      icon={Icon.Clipboard}
+      shortcut={Keyboard.Shortcut.Common.Copy}
+    />
+  );
+};
+
+async function launchCursorRecentProjects(context: { cursorDirectory: { ruleContent: string; replace: boolean } }) {
+  try {
+    await crossLaunchCommand(
+      {
+        name: "index",
+        extensionName: "cursor-recent-projects",
+        ownerOrAuthorName: "degouville",
+        type: LaunchType.UserInitiated,
+        context,
+      },
+      // TODO: callbackLaunchOptions
```

Comment:
I saw you mark this parameter as TODO. Do you have other options that must be passed to the callback launch function?

According to the documentation of [`callbackOptions`](https://github.com/LitoMore/raycast-cross-extension-conventions?tab=readme-ov-file#callbackoptions), the library will automatically set the extension parameters for you.

Reply:
Ah you're absolutely right! This was just some leftover TODO from earlier development that I forgot to clean up. I've removed both the comment and the empty parameter since the callback options are indeed handled automatically. Thanks for catching this!